src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -10324,16 +10321,39 @@ MethodTable * MethodTableBuilder::AllocateNewMT(
 
     MethodTable* pMT = (MethodTable*)(pData + dwGCSize);
 
-    pMT->SetMultipurposeSlotsMask(dwMultipurposeSlotsMask);
+    if (dwNumDicts != 0)
+    {
+        pMT->SetFlag(MethodTable::enum_flag_HasPerInstInfo);
+    }
+
+    if (bmtVT->pDispatchMapBuilder->Count() > 0)
+        pMT->SetFlag(MethodTable::enum_flag_HasDispatchMapSlot);
```

Comment:
We are checking `pDispatchMapBuilder->Count() > 0` twice and very close to each other. Can they be collapsed?

Reply:
Hmm... Probably. I'll give it a try.

Reply:
Yep Looks like a fine idea. I've collapsed them.